lib/capi/src/lib/capitags.ts

```diff
@@ -0,0 +1,87 @@
+import type { Tag } from '@guardian/content-api-models/v1/tag';
+import { deserialzeTagsResponse } from './deserialize';
+
+export const URL_MAX = 2048;
+
+/**
+ * We need to have <2048 chars in the URI or one of the backends objects.
+ * So, we try to bundle all of the tag IDs into one lookup; if that fails, we split in half and try two; etc.
+ * @param tagIdList the list of tag IDs to resolve
+ * @param capiBaseUrl base URL for CAPI
+ * @param capiKey API key for CAPI
+ * @param tagIdListTail remaining tag IDs to process next iteration. Internal use only.
+ * @param prevUrlList URLs already determined. Internal use only.
+ */
+export function buildUriList(
```

Comment:
nice 👍 
i was curious to check and it suggested on that, that even if `tagIdListTail` 's length is higher than URL_MAX it will still divide in small chunks using recursive way

Reply:
Exactly, it splits in half and recurses if the list is too long